src/category_theory/limits/shapes/wide_pullbacks.lean

```diff
@@ -191,4 +191,166 @@ abbreviation has_wide_pullbacks : Prop :=
+  (w : ∀ j, fs j ≫ arrows j = f)
+
+@[simp]
+lemma lift_π (j : J) : lift f fs w ≫ (π j : wide_pullback _ _ arrows ⟶ _) = fs _ :=
+by { simp, refl }
+
+@[simp]
+lemma lift_base : lift f fs w ≫ base = f :=
+by { simp, refl }
+
+lemma eq_lift_of_comp_eq (g : X ⟶ wide_pullback _ _ arrows) :
+  (∀ j : J, g ≫ π j = fs j) → g ≫ base = f → g = lift f fs w :=
+begin
+  intros h1 h2,
+  apply (limit.is_limit (wide_pullback_shape.wide_cospan B objs arrows)).uniq
+    (wide_pullback_shape.mk_cone f fs $ by exact w),
+  rintro (_|_),
+  { apply h2 },
+  { apply h1 }
+end
+
+lemma hom_eq_lift (g : X ⟶ wide_pullback _ _ arrows) :
+  g = lift (g ≫ base) (λ j, g ≫ π j) (by tidy) :=
+begin
+  apply eq_lift_of_comp_eq,
+  tidy,
+end
+
+@[ext]
+lemma hom_ext (g1 g2 : X ⟶ wide_pullback _ _ arrows) :
+  (∀ j : J, g1 ≫ π j = g2 ≫ π j) → g1 ≫ base = g2 ≫ base → g1 = g2 :=
+begin
+  intros h1 h2,
+  rw hom_eq_lift g2,
+  apply eq_lift_of_comp_eq,
+  assumption'
+end
+
+end wide_pullback
+
+namespace wide_pushout
+
+variables {C} {B : C} {objs : J → C} {arrows : Π (j : J), B ⟶ objs j}
+variables [has_wide_pushout B objs arrows]
+
+/-- The `j`-th inclusion to the pushout. -/
+noncomputable
+abbreviation ι (j : J) : objs j ⟶ wide_pushout _ _ arrows :=
+colimit.ι (wide_pushout_shape.wide_span _ _ _) (option.some j)
+
+/-- The unique map from the head to the pushout. -/
+noncomputable
+abbreviation head : B ⟶ wide_pushout B objs arrows :=
+colimit.ι (wide_pushout_shape.wide_span _ _ _) option.none
+
+@[simp]
+lemma head_ι (j : J) : arrows j ≫ (ι j : _ ⟶ wide_pushout B objs arrows) = head :=
+by apply (colimit.cocone (wide_pushout_shape.wide_span B objs arrows)).w
+  (wide_pushout_shape.hom.init j)
+
+/-- Descend a collection of morphisms to a morphism from the pushout. -/
+noncomputable
+abbreviation desc {X : C} (f : B ⟶ X) (fs : Π (j : J), objs j ⟶ X)
+  (w : ∀ j, arrows j ≫ fs j = f) : wide_pushout _ _ arrows ⟶ X :=
+colimit.desc (wide_pushout_shape.wide_span B objs arrows)
+  (wide_pushout_shape.mk_cocone f fs $ by exact w)
+
+variables {X : C} (f : B ⟶ X) (fs : Π (j : J), objs j ⟶ X)
+  (w : ∀ j, arrows j ≫ fs j = f)
+
+@[simp]
+lemma ι_desc (j : J) : (ι j : _ ⟶ wide_pushout _ _ arrows) ≫ desc f fs w = fs _ :=
+by { simp, refl }
+
+@[simp]
+lemma head_desc : head ≫ desc f fs w = f :=
+by { simp, refl }
+
+lemma eq_desc_of_comp_eq (g : wide_pushout _ _ arrows ⟶ X) :
+  (∀ j : J, ι j ≫ g = fs j) → head ≫ g = f → g = desc f fs w :=
+begin
+  intros h1 h2,
+  apply (colimit.is_colimit (wide_pushout_shape.wide_span B objs arrows)).uniq
+    (wide_pushout_shape.mk_cocone f fs $ by exact w),
+  rintro (_|_),
+  { apply h2 },
+  { apply h1 }
+end
+
+lemma hom_eq_desc (g : wide_pushout _ _ arrows ⟶ X) :
+  g = desc (head ≫ g) (λ j, ι j ≫ g) (λ j, by { rw ← category.assoc, simp }) :=
+begin
+  apply eq_desc_of_comp_eq,
+  tidy,
+end
+
+@[ext]
+lemma hom_ext (g1 g2 : wide_pushout _ _ arrows ⟶ X) :
+  (∀ j : J, ι j ≫ g1= ι j ≫ g2) → head ≫ g1 = head ≫ g2 → g1 = g2 :=
```

Comment:
```suggestion
  (∀ j : J, ι j ≫ g1 = ι j ≫ g2) → head ≫ g1 = head ≫ g2 → g1 = g2 :=
```
You could probably also prove this with `colimit.hom_ext` rather than going via `hom_eq_desc`

Reply:
Done. Proof now uses `(co)limit.hom_ext`.